src/class-user-interface.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Duplicate Post user interface.
+ *
+ * @package Duplicate_Post
+ */
+
+namespace Yoast\WP\Duplicate_Post;
+
+/**
+ * Represents the Duplicate Post User Interface class.
+ */
+class User_Interface {
+
+	/**
+	 * Holds the post.
+	 *
+	 * @var \WP_Post
+	 */
+	private $post = null;
+
+	/**
+	 * Holds the global `$pagenow` variable's value.
+	 *
+	 * @var string
+	 */
+	private $pagenow;
+
+	/**
+	 * Holds the object to manage the row actions for the post.
+	 *
+	 * @var Row_Action
+	 */
+	protected $rewrite_and_republish_row_action;
+
+	/**
+	 * Holds the object to create the action link to duplicate.
+	 *
+	 * @var Link_Builder
+	 */
+	protected $link_builder;
+
+	/**
+	 * Admin notices object.
+	 *
+	 * @var Admin_Notices
+	 */
+	private $admin_notices;
+
+	/**
+	 * Initializes the class.
+	 */
+	public function __construct() {
+		global $pagenow;
+		$this->pagenow = $pagenow;
+
+		$this->link_builder                     = new Rewrite_Republish_Link_Builder();
+		$this->rewrite_and_republish_row_action = new Row_Action( $this->link_builder );
+		$this->admin_notices                    = new Admin_Notices();
+
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_scripts' ] );
+		\add_action( 'admin_enqueue_scripts', [ $this, 'should_previously_used_keyword_assessment_run' ], 9 );
+
+		\add_filter( 'post_row_actions', [ $this->rewrite_and_republish_row_action, 'add_action_link' ], 11, 2 );
+		\add_filter( 'page_row_actions', [ $this->rewrite_and_republish_row_action, 'add_action_link' ], 11, 2 );
```

Comment:
Question: `$this->rewrite_and_republish_row_action` is a new Row_Action(), which already has its own register_hooks method called in its constructor and already adds this filters. Is it necessary to re-add these filters here?

Reply:
No, leftovers from a previous commit, I'll remove them